torchci/clickhouse_queries/cost_job_per_job_name/query.sql

```diff
@@ -0,0 +1,22 @@
+select
+    DATE_TRUNC(
+        {granularity: String},
+        rc.date
+    ) as granularity_bucket,
+    rc.job_name as job_name,
+    sum(rc.cost) as total_cost
+from
+    misc.runner_cost rc final
+where
+    rc.date > {startTime: DateTime64(9)}
+    and rc.date < {stopTime: DateTime64(9)}
+    and rc.cost > 0
+    and rc.group_repo in {selectedRepos: Array(String)}
+    and rc.gpu in {selectedGPU: Array(UInt8)}
+    and rc.os in {selectedPlatforms: Array(String)}
+    and rc.provider in {selectedProviders: Array(String)}
+group by
+    granularity_bucket,
+    job_name
+order by
+    granularity_bucket asc
```

Comment:
afaict the only thing that differs in some of the queries is what you group by, so if you want fewer queries, you could do a case statement to determine what to group by

Reply:
Good point, although I haven't been able to write the query in a way that CH accepts it yet.  If doable, this would reduce the number of queries a lot